### PR TITLE
`pallet-updater` tests & benchmarks

### PR DESCRIPTION
**Pull request by Chubek**
Currently 6 tests have been added. All tests passing.

There is a problem that I think will be better for the issues section, but `System::event()` is for some reason entirely empty, making me unable to write event tests.

The tests are:

1. `test_genesis_work`
2. `test_add_member_work`
3. `test_remove_member_work`
4. `test_propose_code_work`
5. `test_vote_code_work`
6. `test_close_vote_work`

### CODE REVIEW

File: bin/node/pallets/pallet-updater/src/tests.rs [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+	BuildStorage,
+};
+
+use crate::{self as pallet_updater, Config};
+
+type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<Test>;
+type Block = frame_system::mocking::MockBlock<Test>;
+
+// Configure a mock runtime to test the pallet.
+frame_support::construct_runtime!(
```
Comment (charmitro) [inline bin/node/pallets/pallet-updater/src/tests.rs:17]: `construct_runtime!` along with the config implementations should be in the `mock.rs` file. We should always follow the structure of others pallets.

**Review by charmitro — CHANGES_REQUESTED** [review]

File: bin/node/pallets/pallet-updater/src/tests.rs [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+	pub const UpdaterMotionDuration: u64 = 0;
+}
+
+impl pallet_updater::Config for Test {
+	type Event = Event;
+	type Call = Call;
+	type MaxProposals = MaxProposalCnt;
+	type MaxMembers = MaxMemberCnt;
+	type MotionDuration = UpdaterMotionDuration;
+}
```
Comment (charmitro) [inline bin/node/pallets/pallet-updater/src/tests.rs:71]: This code should be in its own file called `mock.rs` just like we do for the other pallets.

**Review by charmitro — CHANGES_REQUESTED** [review]

**Comment by zycon91** [comment]
Shouldn't we include some ```should_fail``` tests to check if the errors are working as intended? I think you have "included" those errors check in your ```should_work``` tests with the ```assert_noop!``` you are using. I think we should separate the tests to ```should_work``` and ```should_fail```.

Example of a ```should_fail``` test:
https://github.com/CherryNetwork/Cherry-Node/blob/fc9a103a101540679f1e4e340f3d5433a1192f5b/frame/treasury/src/tests.rs#L246-L251

**Comment by charmitro** [comment]
> Shouldn't we include some `should_fail` tests to check if the errors are working as intended? I think you have "included" those errors check in your `should_work` tests with the `assert_noop!` you are using. I think we should separate the tests to `should_work` and `should_fail`.
> 
> Example of a `should_fail` test:
> 
> https://github.com/CherryNetwork/Cherry-Node/blob/fc9a103a101540679f1e4e340f3d5433a1192f5b/frame/treasury/src/tests.rs#L246-L251

Totally agree with this @Chubek

**Review by zycon91 — APPROVED** [review]
Looks a lot cleaner now!

**Review by charmitro — COMMENTED** [review]
@zycon91 Make sure to test if these work as intendet. I will merge on your call.

**Review by charmitro — APPROVED** [review]

**Review by charmitro — APPROVED** [review]

**Pull request merged**